src/main/java/org/terasology/core/world/generator/facetProviders/SimplexBaseSurfaceProvider.java

```diff
@@ -35,13 +35,21 @@
 @Requires(@Facet(SeaLevelFacet.class))
 public class SimplexBaseSurfaceProvider implements FacetProvider {
     private static final int SAMPLE_RATE = 4;
+    private static final float BEACH_STEEPNESS = 0.05f;
+    private static final float OCEAN_FLOOR_CUTOFF = 0.1f;
 
     private SubSampledNoise surfaceNoise;
 
     @Override
     public void setSeed(long seed) {
-        BrownianNoise source = new BrownianNoise(new SimplexNoise(seed), 8);
-        surfaceNoise = new SubSampledNoise(source, new Vector2f(0.004f, 0.004f), SAMPLE_RATE);
+        float spawnHeight = -1;
+        long currentSeed = (seed % 2 == 0) ? seed - 1 : seed;
+        while (spawnHeight < 0 || spawnHeight > 0.2) {
+            BrownianNoise source = new BrownianNoise(new SimplexNoise(currentSeed), 8);
+            surfaceNoise = new SubSampledNoise(source, new Vector2f(0.0002f, 0.0002f), SAMPLE_RATE);
+            spawnHeight = surfaceNoise.noise(0, 0);
+            currentSeed *= 3;
+        }
```

Comment:
🤔 This makes the implicit assumption that the spawn point is always at (0, 0), right? To be honest, I don't like this (implicit) connection, and I would rather see the responsibility of finding a suitable spawn point somewhere else...
I'm not sure how to proceed with this, though... My gut feeling says that this should not be in here...

Reply:
This makes the assumption that the spawn point is at least near to (0,0). When oceans can be many hundreds of blocks wide, spawning on a tiny platform in the middle of an ocean would be very undesirable. Having a very large search area for a suitable spawn point would be possible, but I can't think of a neat and efficient way to do it.

Reply:
... which leads me to this question: do we really want oceans that are hundreds of blocks wide? There is nothing to for players in that area from a gameplay perspective (yet), and we don't even have boats...

Reply:
The world previews are often very slow themselves, whereas this method is very fast and easy. In most cases, the position of the spawn point doesn't actually matter, so it is very likely to be (0,0). Even if it isn't, there's no harm in having this here except the slight additional complexity, which is much less than the complexity of something to search for a spawn point elsewhere would be.

I don't have much of an argument in favour of the large oceans, but I think it's nice. I like that there are actually distinct, well-separated continents, and that large islands exist rather than practically all of the land being connected together. I guess it also just provides more variation.